### ISSUE TITLE
Built in Updates

### ISSUE THREAD

**Issue by gohanman** [CORE-POS/IS4C#347]
The proposed update mechanism uses a little script I put together, giterate.
https://github.com/gohanman/giterate

It's in python instead of PHP because the options for git bindings/libraries were better.

giterate does a few things:
1. Adds a read-only remote for itself to the repository (to avoid auth issues if the default remote is password protected)
2. Fetches (git-fetch) via the remote to get new tags and commits
3. Parses and sorts version tags to determine the "newest" one
4. Merges (git-merge) the newest tag on request

It's a command line tool that describes its options pretty well, I think. The configuration file is optional but specifying paths and remotes makes the commands a bit long. A config file for CORE would look like this:
```
[CORE POS]
path=/path/to/local/repo
url=https://github.com/CORE-POS/IS4C
```
Checking for updates would be:
```
python giterate.py --name "CORE POS"
```
Merging the latest tag would be:
```
python giterate.py --name "CORE POS" --update
```
A merge conflict should, in theory, result in an automatic rollback to the previous state unless the --force option is specified.

I added giterate to CORE as an optional submodule. It needs to be installed that way if using Fannie's integrated update checking task, but it could be installed separately and used manually or via other custom scripting.

The primary barrier to having a built-in task *apply* updates as well as check what's available is the web server user would need write-access to the entire tree. It's not uncommon to have that kind of setup - e.g., Wordpress updating itself - but from a security standpoint I don't think it's ideal.

**Comment by gohanman** [comment]
Re-do & change of direction:
eb96d51479423e6e12b06c041129cc512e500490

Fannie reads available versions directly using github's web API. There's no dependency to simply check for new versions and tell the user they're available.

When a new version is available, the update page provides detailed git instructions for fetching and applying the update and troubleshooting problems. This is probably less work overall. Setting up python and related packages, git submodules, and a task to run the python version check is a whole lot of up front effort for a really minimal gain in simplifying the fetch+merge step.

**Comment by gohanman** [comment]
Revised approach included in 1.1.0
